source/develop/release-management/features/network/autodefine-external-network.md

```diff
@@ -0,0 +1,134 @@
+  network to one of it's interfaces. Same role applies for hotplug of VM 
+  interfaces. 
+
+## Detailed description
+
+### Creating of new physical network
+
+When creating new network choose its name. Check the `VM network`. If vlan 
+tagging is needed, check the `Enable VLAN tagging` and put the tag into the 
+field. Please note that auto-define can be triggered only on top of network that
+is not marked as external. 
+
+![Creation of physical network](../../../../images/wiki/auto-define_create.png  "New physical network")
+
+After confirmation if all conditions were met, the external network connected to 
+the physical one, will be created simultaneously. The auto-defined external 
+network is named `external_name_of_physical_network` (The physical network name
+might be shortened so that auto-defined network name is not bigger than 256 
+characters). 
+
+![Auto-define network created](../../../../images/wiki/auto-define_created.png  "Auto-define external network")
+
+In case there are clusters with different default external providers in the data 
+center the Engine will create multiple external networks for every provider and 
+add a number behind to this auto-defined network name. 
+
+Please note that all hosts running VMs connected to the external network has
+to be attached to the physical network. 
+
+
+
+## Implementation
+
+### Engine
+
+* A new query `GetAllVmNetworksByClusterId` that returns only VM networks is 
+  added.For Cluster with OVS switch type, this query returns only external 
+  networks. 
+
+* Validations for all limited operations with auto-defined external network and 
+  its linked physical network. According to [Autodefine External Network - Limitations](#limitations)
+
+* Added new command `AutodefineExternalNetwork`. This command is 
+  triggered every time a new network is created. The command validates the 
+  incoming network and its parameters. Then it finds all clusters with OVS 
+  switch type attached to incoming network and clusters external providers.
+  The command continues with creating new external network and connecting it to 
+  the incoming physical network and attaching to clusters. The command might 
+  create multiple external networks, one for each external provider in the data 
+  center. 
+  
+* Added new function `getAllExternalNetworksLinkedToPhysicalNetwork` into 
+  `NetworkDao` which returns list of all external networks linked to provided
+  physical network. 
+
+
+## Testing
+
+Testing should cover: 
+
+* Addition/Removal new physical network.
+
+* Physical network linked to external network cannot be 
+  removed.
+
+* Only external networks can be used for VM interfaces.
+
+* Cluster switch type cannot be updated to OVS for existing cluster.
+
+* VMs cannot be scheduled on hosts without attached physical network linked to 
+  auto-define network. 
+
+* Interface cannot be hotplugged on hosts without attached physical network 
+  linked to auto-define network.
```

Comment:
I expect tests for the scheduling, too.

Reply:
There is paragraph for scheduling testing

Reply:
indeed